src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -587,11 +601,139 @@ data class InteractiveTxSession(
             }
         }
 
-        return InteractiveTxSessionAction.SignSharedTx(sharedTx, sharedOutputIndex, txComplete)
+        return InteractiveTxSessionAction.SignSharedTx(sharedTx, txComplete)
     }
 
     companion object {
         // We restrict the number of inputs / outputs that our peer can send us to ensure the protocol eventually ends.
         const val MAX_INPUTS_OUTPUTS_RECEIVED = 4096
     }
 }
+
+sealed class InteractiveTxSigningSessionAction {
+    /** Wait for their tx_signatures before sending ours. */
+    object WaitForTxSigs : InteractiveTxSigningSessionAction()
+    /** Send our tx_signatures: we cannot forget the channel until it has been spent or double-spent. */
+    data class SendTxSigs(val fundingTx: LocalFundingStatus.UnconfirmedFundingTx, val commitment: Commitment, val localSigs: TxSignatures) : InteractiveTxSigningSessionAction()
+    data class AbortFundingAttempt(val reason: ChannelException) : InteractiveTxSigningSessionAction() {
+        override fun toString(): String = reason.message
+    }
+}
+
+/**
+ * Once a shared transaction has been created, peers exchange signatures for the commitment and the shared transaction.
+ * We store the channel state once we reach that step. Once we've sent tx_signatures, we cannot forget the channel
+ * until it has been spent or double-spent.
+ */
+data class InteractiveTxSigningSession(
+    val fundingParams: InteractiveTxParams,
+    val fundingTx: PartiallySignedSharedTransaction,
+    val localCommitSig: CommitSig,
```

Comment:
I believe `localCommitSig` should be returned as an action `InteractiveTxSigningSessionAction.SendCommitSig` by the `create` method.